core/src/service/document_service.rs

```diff
@@ -4,15 +4,13 @@ use lockbook_shared::crypto::DecryptedDocument;
 use lockbook_shared::document_repo;
 use lockbook_shared::file_like::FileLike;
 use lockbook_shared::file_metadata::FileType;
-use lockbook_shared::tree_like::{TreeLike, TreeLikeMut};
+use lockbook_shared::tree_like::TreeLike;
 use uuid::Uuid;
 
 impl<Client: Requester> RequestContext<'_, '_, Client> {
     pub fn read_document(&mut self, id: Uuid) -> CoreResult<DecryptedDocument> {
-        let mut tree = self
-            .tx
-            .base_metadata
-            .stage(&mut self.tx.local_metadata)
+        let mut tree = (&self.tx.base_metadata)
+            .stage(&self.tx.local_metadata)
```

Comment:
(here and elsewhere) references to `self.tx.x_metadata` are used to indicate whether they may be mutated by an operation